core/domain/exp_services.py

```diff
@@ -565,6 +570,194 @@ def apply_change_list(exploration_id, change_list):
         python_utils.reraise_exception()
 
 
+def calculate_exploration_size_using_proto(exploration):
```

Comment:
I'd expect the Exploration & constituent domain objects to have functions like toAndroidProto to convert the corresponding object to proto, and then just calling the ByteSize() method on the converted proto object here. That seems more in-line with our other data conversion patterns (particularly to/from dicts which we can then generically convert to/from json).

Reply:
ah, yes I should have created it in that way. Update now. 
so, now we have `to_android_proto` function inside the Exploration domain class and anyone can pass the exploration domain object and get the android proto. 

Also, we can pass the dict too, something like `to_android_proto(exploration.to_dict())`, in this case, we have use this dictionary to set our proto inside `to_android_proto`.